rust/main/chains/hyperlane-ethereum/src/rpc_clients/provider.rs

```diff
@@ -93,6 +93,9 @@ pub trait EvmProviderForSubmitter: Send + Sync {
 
     /// Send transaction into blockchain
     async fn send(&self, tx: &TypedTransaction, function: &Function) -> ChainResult<H256>;
+
+    /// Read-only call into blockchain
+    async fn call(&self, tx: &TypedTransaction, function: &Function) -> ChainResult<bool>;
```

Comment:
hm, this assumes the return type of the call is always `bool`. Should we rename this method to reflect that? Since the `etc_call` rpc could return anything, and when I initially read this I thought that's what it did